executor/write.go

```diff
@@ -200,21 +200,27 @@ func (e *DeleteExec) deleteMultiTables() error {
 			break
 		}
 
-		for _, entry := range joinedRow.RowKeys {
-			if !isMatchTableName(entry, tblMap) {
-				continue
-			}
-			if tblRowMap[entry.Tbl] == nil {
-				tblRowMap[entry.Tbl] = make(map[int64][]types.Datum)
+		for id, cols := range e.SelectExec.Schema().TblID2handle {
+			tbl := e.tblID2Table[id]
+			for _, col := range cols {
```

Comment:
Indeed, the join order may be changed. Like `update a as b, a as c`, the physical plan may be `c index join b`